pkg/app/middlewares/instance_middleware.go

```diff
@@ -0,0 +1,57 @@
+package middlewares
+
+import (
+	"context"
+	"strings"
+
+	"github.com/aserto-dev/aserto-grpc/grpcutil"
+	"github.com/aserto-dev/go-authorizer/aserto/authorizer/v2"
+	"github.com/aserto-dev/go-authorizer/aserto/authorizer/v2/api"
+	"github.com/aserto-dev/topaz/pkg/cc/config"
+	grpcmiddleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	"github.com/rs/zerolog"
+	"google.golang.org/grpc"
+)
+
+type InstanceMiddleware struct {
```

Comment:
I suggest renaming this to PolicyInstanceMiddleware, as InstanceMiddleware is very opaque, instance of what?

I would also rename the file to just policy_instance or policy as middleware is implied by the directory name.